public/app/core/components/RolePicker/TeamRolePicker.tsx

```diff
@@ -15,19 +15,26 @@ export interface Props {
 export const TeamRolePicker: FC<Props> = ({ teamId, orgId, roleOptions, disabled, builtinRolesDisabled }) => {
   const [appliedRoles, setAppliedRoles] = useState<Role[]>([]);
 
-  const { loading } = useAsync(async () => {
+  const getTeamRoles = async () => {
     try {
       const teamRoles = await fetchTeamRoles(teamId, orgId);
       setAppliedRoles(teamRoles);
     } catch (e) {
       // TODO handle error
       console.error('Error loading options');
     }
-  }, [orgId, teamId]);
+  };
+
+  const { loading } = useAsync(getTeamRoles, [orgId, teamId]);
```

Comment:
This is not really how this hook is supposed to be used, it already handles the state internally, stored onto `value` property - https://github.com/streamich/react-use/blob/master/docs/useAsync.md. To get advantage of that, we just need to return team roles inside of `getTeamRoles` instead of setting them to state. With this approach saving applied roles to state becomes redundant. If we need to do a refetch, there's a separate hook for that - [useAsyncFn](https://github.com/streamich/react-use/blob/master/docs/useAsyncFn.md).

Reply:
In case of `useAsyncFn()` component might look like this. But I do not like that we need to call it additionally, for example in the `useEffect()` hook. Do you see shorter solution?
```ts
const [{ loading, value: appliedRoles }, getUserRoles] = useAsyncFn(async () => {
    try {
      if (contextSrv.hasPermission(AccessControlAction.ActionUserRolesList)) {
        return await fetchUserRoles(userId, orgId);
      }
    } catch (e) {
      // TODO handle error
      console.error('Error loading options');
    }
    return [];
  }, [orgId, userId]);

  useEffect(() => {
    getUserRoles();
  }, [orgId, userId, getUserRoles]);

  const onRolesChange = async (roles: string[]) => {
    await updateUserRoles(roles, userId, orgId);
    await getUserRoles();
  };
```

Reply:
We shouldn't need to call it in `useEffect`, `useAsyncFn` should do that automatically on component mount.

Reply:
Hm, but it does not do it.

Reply:
Weird, yeah, it doesn't call it on mount. I think in this case it's easier just to make an additional call in `useEffect`. `useAsyncFn` accepts a third param, which is an async function that returns the initial state, but that's a bit complicated to get right.

Reply:
I think the initial state param matches `useAsyncFn` use case - run func on action, so it needs some initial state. In case of `useAsync` initial state is unnecessary since it runs on mount.